cranelift/jit/src/memory.rs

```diff
@@ -56,10 +58,14 @@ impl PtrLen {
         // Safety: We assert that the size is non-zero above.
         let ptr = unsafe { alloc::alloc(layout) };
 
-        Ok(Self {
-            ptr,
-            len: alloc_size,
-        })
+        if !ptr.is_null() {
+            Ok(Self {
+                ptr,
+                len: alloc_size,
+            })
+        } else {
+            Err(io::Error::last_os_error())
```

Comment:
This may not necessarily be correct. Not every global allocator impl sets errno. In fact I believe most don't.

Reply:
Ah, I didn't realize. Maybe it would be better to just return `io::Error::from(io::ErrorKind::OutOfMemory)` ourselves here, then?

Reply:
Yeah, that makes sense.

Reply:
Updated to directly return `OutOfMemory` instead of trying to use `last_os_error()`